AsyncDisplayKit/Details/ASEnvironment.h

```diff
@@ -154,13 +156,15 @@ ASDISPLAYNODE_EXTERN_C_END
   [super setEnvironmentState:environmentState];\
   ASEnvironmentTraitCollection currentTraits = environmentState.traitCollection;\
   if (ASEnvironmentTraitCollectionIsEqualToASEnvironmentTraitCollection(currentTraits, oldTraits) == NO) {\
-    NSArray<NSArray <ASCellNode *> *> *completedNodes = [self.view.dataController completedNodes];\
-    for (NSArray *sectionArray in completedNodes) {\
-      for (ASCellNode *cellNode in sectionArray) {\
-        ASEnvironmentStatePropagateDown(cellNode, currentTraits);\
-        [cellNode setNeedsLayout];\
+    dispatch_async(dispatch_get_main_queue(), ^{\
```

Comment:
The main issue is that I am accessing `[self.view.dataController completedNodes]` which must be on a main thread. Should I get that value on the main thread, then jump back to a bg thread?